### ISSUE TITLE
Create the custom IAM Role in linked account 

### ISSUE THREAD

**Issue by 20FS21** [awslabs/aws-deployment-framework#316]
Hello team,
I would like to add the follow IAM Custom Role into my all Linked account in AWS Organization:

#APPEND THIS CONTENT TO global.yaml FILE
  AutoRemediationIamRole:
    Type: "AWS::IAM::Role"
    Properties:
      RoleName: adf-remediation-role
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - ec2.amazonaws.com
                - events.amazonaws.com
                - ssm.amazonaws.com
            Action:
              - "sts:AssumeRole"
      ManagedPolicyArns:
        - "arn:aws:iam::aws:policy/service-role/AmazonSSMAutomationRole"
        - "arn:aws:iam::aws:policy/AmazonEC2FullAccess"

  AutomationPassRolePolicy:
    Type: "AWS::IAM::Policy"
    Properties:
      PolicyName: passAutomationRole
      PolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Action:
              - "iam:PassRole"
            Resource: !GetAtt AutoRemediationIamRole.Arn
      Roles:
        - !Ref AutoRemediationIamRole

 For do this, could I add this Role in the end of the file global.yml in the adf-bootstrap.
Correct?

**Comment by sbkok** [comment]
Hi @20FS21,

I don't recommend that. The `adf-bootstrap/global.yml` is used for ADF resources specifically.
If you upgrade to a newer version of ADF it gets overwritten.

For roles that are required for ADF pipelines, you can use `adf-bootstrap/global-iam.yml` instead.
However, the `adf-bootstrap/global-iam.yml` file is only applied for accounts which don't have a specific OU folder, like the deployment account for example. So you would also need to change it in the `adf-bootstrap/deployment/global-iam.yml` file, and others, if you want to include literally all your accounts. You can [find an example here](https://github.com/awslabs/aws-deployment-framework/blob/master/src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/example-global-iam.yml)

**However**, deploying this way is far from ideal as it is not deployed using a pipeline, if it fails the errors are harder to trace and fix, plus all new pipeline definitions will be blocked until you fixed the error.

Instead, you should only add roles and policies to the `global-iam.yml` files that are required by ADF to operate the pipelines. For example, to define what CloudFormation is allowed to manage in an account.

I would recommend to deploy any other IAM role, policy or other resource for that matter using a pipeline itself.
Such that you can manage the pipeline, monitor it and deploy resources in stages across all your accounts. Starting with the sandbox / development accounts and moving upwards towards production accounts in stages.

You can target the accounts using [OUs](https://github.com/awslabs/aws-deployment-framework/blob/master/docs/user-guide.md#deployment-map) or [tags](https://github.com/awslabs/aws-deployment-framework/blob/master/docs/user-guide.md#targeting-via-tags) for example. The links point to the section in the documentation describing this.

TLDR; use pipelines instead of the changing files in the adf-bootstrap repository.

Best regards, Simon

**Comment by 20FS21** [comment]
Thanks a lot.
In my deployment_map.yml I use: 

- name: XXXXXX
   default_providers:
     source:
       provider: codecommit
       properties:
         account_id: XXXXXX
         repository: repository-1
     build:
       provider: codebuild
     deploy:
       provider: cloudformation
   params:
     notification_endpoint: email@email.com
   regions:
     - eu-central-1
   targets:
     - path: [XXXXXX]
       name: xxxxx-changeset
       properties:
         stack_name: xxxxx
         action: CHANGE_SET_REPLACE
     - approval
     - path: [xxxxxxx] 
       name: xxxxxx-apply
       properties:
         stack_name: XXXXXXXX

When you say:
"
I would recommend to deploy any other IAM role, policy or other resource for that matter using a pipeline itself.
Such that you can manage the pipeline, monitor it and deploy resources in stages across all your accounts. Starting with the sandbox / development accounts and moving upwards towards production accounts in stages.
"
it means that i need to change the parto of pipeline in deployment_map.yml or another please?

Thanks a lot

**Comment by sbkok** [comment]
Hi @20FS21,

Apologies for the late reply.
I would recommend to setup a separate pipeline for the IAM, like documented [in the Deployment Map section of the user docs of ADF here](https://github.com/awslabs/aws-deployment-framework/blob/master/docs/user-guide.md#deployment-map).

So you create a separate pipeline for the IAM parts and add another a new pipeline per project/stack/repo you want to deploy.
As described in the [Additional Deployment Maps section](https://github.com/awslabs/aws-deployment-framework/blob/master/docs/user-guide.md#additional-deployment-maps), you can create separate deployment map files and bundle the pipeline definitions together in a way that works best for you.
I hope this helps.

Best regards, Simon

**Comment by sbkok** [comment]
I'm closing this issue as it has been inactive for a few weeks. This probably means that it has been fixed in the meanwhile.

Please reopen if you still encounter this issue with the [latest stable version](https://github.com/awslabs/aws-deployment-framework/releases).

Thank you!

**Comment by stefl3n** [comment]
I have a similar problem, because I have a custom automation in the deployment account (not inside adf) which needs a specific role with certain policy to automate all.
Can you explain in detail which is the best way to create a new role and new policy in certain OUs or in all Accounts using ADF although this role and its policy are independent from adf?

Thanks,

Stefano
